server/controllers/profileController.js

```diff
@@ -23,6 +23,17 @@ const getProfile = async (req, res) => {
     return res.status(404).json({ error: "No such Profile." });
   }
 
+  // check if profile is anonymous
+  if (profile.anonymous) {
+    // do not send sensitive information
+    profile.linkedin = "";
+    profile.pfp = "";
+    profile.location = "";
+
+    profile.firstName = "Anonymous";
+    profile.lastName = "";
+  }
+
```

Comment:
Optional but here's a little pieece of JavaScript magic for doing this more elegantly!

```ts
if (profile.anonymous) {
    const { linkedin, pfp, location, ...anonymousProfile } = profile;
    profile = {
      ...anonymousProfile,
      firstName: "Anonymous",
      lastName: ""
    };
  }
```

Reply:
I love.